### PR TITLE
add test/presubmit-tests.sh script required by prow

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

File: test/presubmit-tests.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+function unit_tests() {
+  echo "Running unit tests"
+  make -C ../ test
+}
+
+function integration_tests() {
+  header "Running integration tests"
+  echo "TODO(aaron-prindle): add integration tests"
```
Comment (bobcatfish) [inline test/presubmit-tests.sh]: you could tag https://github.com/knative/build-pipeline/issues/16 , doesn't have to be on your shoulders necessarily!

**Comment by bobcatfish** [comment]
Thanks @aaron-prindle ! My last request: are you interested in trying to write this in go or python? 😇 If not we can go ahead with this but I'd be curious to see what this script looks like in one of those langs, might be a bit easier to work with

**Comment by aaron-prindle** [comment]
I don't mind re-writing it but I want to see prow successfully run with something simple first.  I agree that bash scripts can get pretty unwieldy but for now I image this bash script will just call our makefile and i can't really foresee it ever doing more than that (*knock on wood).

**Comment by bobcatfish** [comment]
> I don't mind re-writing it but I want to see prow successfully run with something simple first. I agree that bash scripts can get pretty unwieldy but for now I image this bash script will just call our makefile and i can't really foresee it ever doing more than that (*knock on wood).

kk! the first time we have to change it i vote we re-write it XD

**Comment by bobcatfish** [comment]
and so it begins  - we need to add `/lgtm` to our contributor docs asap, I'll take care of it!

**Comment by bobcatfish** [comment]
and... approved I guess..

/approve

**Comment by bobcatfish** [comment]
wait a sec, we've got tide but not prow, i guess i need to add the labels manually 🤔

File: test/presubmit-tests.sh (hunk 1 above)
Comment (adrcunha) [reply test/presubmit-tests.sh]: +1

File: test/presubmit-tests.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  make test
+}
+
+function integration_tests() {
+  header "Running integration tests"
+  echo "TODO(aaron-prindle): add integration tests"
+}
+
+# Process flags and run tests accordingly.
+function main() {
```
Comment (adrcunha) [inline test/presubmit-tests.sh]: You're unnecessarily duplicating code that is now intended to be vendored. I strongly recommend that you follow this pattern instead:

https://github.com/knative/docs/blob/master/test/presubmit-tests.sh

**Pull request merged**